Game/MyProject/Game/RoadGenerator/RoadEditor.cpp

```diff
@@ -0,0 +1,219 @@
+        {
+            bezierMarginPoints[i].first->remove();
+            bezierMarginPoints[i].second->remove();
+        }
+        bezierMarginPoints.clear();
+        
+        
+        for(int i = 0; i < rMiddlePoints.size(); i++)
+        {
+            std::shared_ptr<SceneNode> point = std::make_shared<SceneNode>("bezierMiddlePoint" + std::to_string(i));
+            sceneGraph->getRoot()->addChild(point);
+            point->setLocalPosition(rMiddlePoints[i]);
+            point->setRenderable(meshInstanceBlue);
+            bezierMiddlePoints.push_back(point);
+        }
+        
+        for(int i = 0; i < rMarginPoints.size(); i++)
+        {
+            std::shared_ptr<SceneNode> pointL = std::make_shared<SceneNode>("bezierMarginLPoint" + std::to_string(i));
+            sceneGraph->getRoot()->addChild(pointL);
+            pointL->setLocalPosition(rMarginPoints[i].first);
+            pointL->setRenderable(meshInstanceBlue);
+            
+            std::shared_ptr<SceneNode> pointR = std::make_shared<SceneNode>("bezierMarginRPoint" + std::to_string(i));
+            sceneGraph->getRoot()->addChild(pointR);
+            pointR->setLocalPosition(rMarginPoints[i].second);
+            pointR->setRenderable(meshInstanceBlue);
+            bezierMarginPoints.push_back(std::make_pair(pointL, pointR));
+        }
+        
+        //render bezier points at final to see the color
+        for(int i = 0; i < bezierPoints.size(); i++)
+        {
+            std::shared_ptr<SceneNode> point = std::make_shared<SceneNode>("bezierPoint" + std::to_string(i));
+            sceneGraph->getRoot()->addChild(point);
+            point->setLocalPosition(bezierPoints[i]);
+            
+            if(i % 3 == 0)
+                point->setRenderable(meshInstanceRed);
+            else
+                point->setRenderable(meshInstanceGreen);
+            
+            bezierScenePoints.push_back(point);
+        }
+        
+    }
+}
+
+void RoadEditor::buildRoadEditor()
+{
+    resizeRoadEditor();
+    
+    for(int i = 0; i < bezierPoints.size(); i++)
+        bezierScenePoints[i]->setLocalPosition(bezierPoints[i]);
+    
+    for(int i = 0; i < rMiddlePoints.size(); i++)
+        bezierMiddlePoints[i]->setLocalPosition(rMiddlePoints[i]);
+    
+    for(int i = 0; i < rMarginPoints.size(); i++)
+    {
+        bezierMarginPoints[i].first->setLocalPosition(rMarginPoints[i].first);
+        bezierMarginPoints[i].second->setLocalPosition(rMarginPoints[i].second);
+    }
+    
+    //coloring stuff
+    for(int i = 0; i < bezierScenePoints.size(); i++)
+    {
+        if(i % 3 == 0)
+            bezierScenePoints[i]->setRenderable(meshInstanceRed);
+        else
+            bezierScenePoints[i]->setRenderable(meshInstanceGreen);
+    }
+    
+    //set yellow the editable point
+    bezierScenePoints[roadGenerator->getEditIndex()]->setRenderable(meshInstanceYellow);
+}
+
+RoadEditor::~RoadEditor()
+{
+    delete roadGenerator;
```

Comment:
use std::shared_ptr for roadGenerator variable. No need to delete the memory with that.